### PR TITLE
Adding Automation to deploy the template and the other parts needed to send data to GA4

### PR DESCRIPTION
**Pull request by adamread**
Provides a simple website to deploy the template, add the tag, and add the other tags needed to send the CWV data to GA4.

### CODE REVIEW

**Review by demirj — APPROVED** [review]
LGTM

File: automation/gtm_automation.js [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// The OAuth token client used to authenticate API calls.
+let tokenClient;
+
+// the GTM variables we will create. This is augmented with 'attribution' if
+// the attribution build is used.
+const VariableNames = [
```
Comment (victor-paunescu) [inline automation/gtm_automation.js:22]: nit: Use CONSTANT_CASE for global constant values.
See https://google.github.io/styleguide/tsguide.html#naming-rules-by-identifier-type
```suggestion
const VARIABLE_NAMES = [
```

File: automation/gtm_automation.js [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+ * where the document is accessed. This config is then passed down the chain as
+ * the GTM API calls are made.
+ *
+ * The template file is read in here and then passed to createTemplate() along
+ * with the config to start the deployment workflow.
+ */
+function runDeployment() {
+  lockForm();
+  // set up the global config
+  const gtmURL = document.getElementById('gtm-url').value;
```
Comment (victor-paunescu) [inline automation/gtm_automation.js:112]: nit: Should be gtmUrl
See https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
  const gtmUrl = document.getElementById('gtm-url').value;
```

File: automation/gtm_automation.js [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+    if (resp.error !== undefined) {
+      throw (resp);
+    }
+    gapi.client.tagmanager.accounts.containers.workspaces.templates.create({
+      parent: config.get('parent'),
+      templateData: content,
+    })
+      .then((gtmResp) => {
+        addSuccessMessage('Created Web Vitals Template');
+        const templateID = gtmResp.result.templateId;
```
Comment (victor-paunescu) [inline automation/gtm_automation.js:160]: nit: Should be templateId
https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
        const templateId = gtmResp.result.templateId;
```

File: automation/gtm_automation.js [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+ * Creates a tag using the previously installed template to measure CWVs.
+ *
+ * If the tag is successfully created, `createEventTrigger` is called.
+ *
+ * @param config {Map<string, *>} The configuration map.
+ * @param templateID {string} The ID the template was assigned when it was
+ * created.
+ */
+function createTag(config, templateID) {
+  const containerID = config.get('parent').split('/')[3];
```
Comment (victor-paunescu) [inline automation/gtm_automation.js:187]: nit: Should be containerId
https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
  const containerId = config.get('parent').split('/')[3];
```

File: automation/gtm_automation.js [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+/**
+ * Creates a tag using the previously installed template to measure CWVs.
+ *
+ * If the tag is successfully created, `createEventTrigger` is called.
+ *
+ * @param config {Map<string, *>} The configuration map.
+ * @param templateID {string} The ID the template was assigned when it was
+ * created.
+ */
+function createTag(config, templateID) {
```
Comment (victor-paunescu) [inline automation/gtm_automation.js:186]: nit: Should be templateId
https://google.github.io/styleguide/tsguide.html#camel-case
```suggestion
function createTag(config, templateId) {
```

**Review by victor-paunescu — CHANGES_REQUESTED** [review]
LGTM besides these minor comments

File: automation/gtm_automation.js (hunk 5 above)
Comment (adamread) [reply automation/gtm_automation.js]: Done

File: automation/gtm_automation.js (hunk 4 above)
Comment (adamread) [reply automation/gtm_automation.js]: Done

**Review by adamread — COMMENTED** [review]
Code updated.

**Review by victor-paunescu — APPROVED** [review]

**Review by demirj — APPROVED** [review]
LGTM

**Pull request merged**